### PR TITLE
docs: add more info and structure to usage docs

### PR DESCRIPTION
**Pull request by grjte**
This PR updates the usage docs to make the resources that we provide more visible by adding:

- "CLI" and "Development Tooling" to the sidebar separately
- a shorter usage page with a summary of the various resources we have available and links to each of them
  - a link to the playground
  - a link to the scaffolded repo
  - separate links to CLI, REPL, Debugger

One of the goals is to give us a single URL that provides information about all of these resources, which can be linked from the scaffolded repo and the playground - we can now use the usage/main page for that.

### CODE REVIEW

**Review by Dominik1999 — APPROVED** [review]
Super cool. I left only one mini nit.

**Review by frisitano — COMMENTED** [review]
This looks good! I think we should also add a section on `.inputs` files so that uses know how to define and use them.  Something similar to what we have [here](https://github.com/0xPolygonMiden/examples/blob/frisitano-advice-provider-playground/playground/src/markdown/Explainer.md?plain=1#L14-L49)

File: docs/src/intro/usage/main.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+The above functionality is also exposed via the single [miden](https://crates.io/crates/miden) crate, which also provides a CLI interface for interacting with Miden VM.
+
+## Miden VM Development Tools and Resources
+
+The following tools are available for interacting with Miden VM:
+
+* Via the [Miden crate](https://crates.io/crates/miden) (or within the Miden VM repo):
+    * [CLI](cli.md)
+    * [Debugger](development_tooling.md/#miden-debugger)
+    * [REPL](development_tooling.md/#repl)
```
[the remark this answers is not in the document]
Comment (grjte) [reply docs/src/intro/usage/main.md:18]: Oh strange, it ran fine for me. I'll make the update though, thanks!

**Comment by grjte** [comment]
> I think we should also add a section on .inputs files so that uses know how to define and use them. Something similar to what we have [here](https://github.com/0xPolygonMiden/examples/blob/frisitano-advice-provider-playground/playground/src/markdown/Explainer.md?plain=1#L14-L49)

Yeah, this would be a good addition. Are you thinking it should go on the "main" usage page? (Also, I might leave it for a separate PR, since I have a few other things queued and don't want to have this change blocked waiting for that)

**Pull request merged**